src/dhcp.c

```diff
@@ -2320,6 +2320,34 @@ int dhcp_dnsDNAT(struct dhcp_conn_t *conn,
       iph->protocol == PKT_IP_PROTO_UDP &&
       udph->dst == htons(DHCP_DNS)) {
 
+#ifdef ENABLE_SESSDNS
+    if (_options.dns_per_session) {
+      struct app_conn_t *appconn = (struct app_conn_t *)conn->peer;
+      if (appconn) {
+        /* If force DNS1 is set */
+        if (appconn->s_params.dns1.s_addr && (iph->daddr == conn->dns1.s_addr)) {
+          conn->dnatdns = iph->daddr;
+          iph->daddr = appconn->s_params.dns1.s_addr;
+          *do_checksum = 1;
+#if(_debug_)
+          if (_options.debug) syslog(LOG_DEBUG, "DNAT DNS1 -> dns[%s]", inet_ntoa(appconn->s_params.dns1));
+#endif
+        }
+
+         /* If force DNS2 is set */
```

Comment:
But I don't think that they will be any overriding here, if `ip->daddr` match the DNS1, it will not match DNS2.
But we can add a simple `else`, to handle the case if the `dns1 == dns2`.